### PR TITLE
PHOENIX-7397 Optimize ClientAggregatePlan/ClientScanPlan when inner q…

### PR DESCRIPTION
**Pull request by comnetwork**
…uery is UnionPlan

The main modifications are as follows: 
In the `QueryCompiler.compileSingleQuery` method, if it is found that the inner subquery compilation results is a `UnionPlan`, the `UnionCompiler.optimizeUnionOrderByIfPossible `method is introduced to determine whether the inner `UnionPlan` meets the conditions for optimizing the sorting of the outer query：

-  If every subquery in `UnionPlan` is ordered, and `QueryPlan.getOutputOrderBys` of each subquery are equal(absolute equality or the same column name is unnecessary, just column types are compatible and columns count is same), and at the same time the outer query of `UnionPlan` has group by or order by, we would further examine whether
maintaining this order for the entire `UnionPlan` can compile out the outer query's group by or order by. 

- If it is sure, then `UnionPlan` just to perform a simple merge on the output of each subquery to ensure the overall order of the union all;
- otherwise, `UnionPlan` would not perform any special processing on the output of the subqueries.

### CODE REVIEW

File: phoenix-core-client/src/main/java/org/apache/phoenix/parse/OrderByNode.java [hunk 1]
```diff
@@ -84,4 +85,16 @@ public void toSQL(ColumnResolver resolver, StringBuilder buf) {
         if (!orderAscending) buf.append(" DESC");
         if (nullsLast) buf.append(" NULLS LAST ");
     }
+
+    public boolean isLiteral() {
```
Comment (tkhurana) [inline phoenix-core-client/src/main/java/org/apache/phoenix/parse/OrderByNode.java:89]: Maybe we could rename this method to `isIntegerLiteral`

**Review by tkhurana — APPROVED** [review]
One small comment but other than that looks good. Thanks

**Comment by comnetwork** [comment]
@tkhurana, thank you very much for reviewing, already fixed the problem.

File: phoenix-core-client/src/main/java/org/apache/phoenix/parse/OrderByNode.java (hunk 1 above)
Comment (comnetwork) [reply phoenix-core-client/src/main/java/org/apache/phoenix/parse/OrderByNode.java]: Ok, already fixed.

**Pull request merged**